subsys/bluetooth/controller/hci/hci.c

```diff
@@ -1604,6 +1604,41 @@ static void le_big_terminate_sync(struct net_buf *buf, struct net_buf **evt,
 
 #if defined(CONFIG_BT_CONN)
 #if defined(CONFIG_BT_CENTRAL)
+
+static uint8_t check_cconn_params(bool ext, uint16_t scan_interval,
+				  uint16_t scan_window,
+				  uint16_t conn_interval_max,
+				  uint16_t conn_latency,
+				  uint16_t supervision_timeout)
+{
+	if (scan_interval < 0x0004 || scan_window < 0x0004 ||
+	    (!ext && (scan_interval > 0x4000 || scan_window > 0x4000))) {
+		return BT_HCI_ERR_INVALID_PARAM;
+	}
+
+	if (conn_interval_max < 0x0006 || conn_interval_max > 0x0C80) {
+		return BT_HCI_ERR_INVALID_PARAM;
+	}
+
+	if (conn_latency > 0x01F3) {
+		return BT_HCI_ERR_INVALID_PARAM;
+	}
+
+	if (supervision_timeout < 0x000A || supervision_timeout > 0x0C80) {
+		return BT_HCI_ERR_INVALID_PARAM;
+	}
+
+	/* sto * 10ms > (1 + lat) * ci * 1.25ms * 2
+	 * sto * 10 > (1 + lat) * ci * 2.5
+	 * sto * 2 > (1 + lat) * ci * 0.5
+	 */
+	if ((supervision_timeout << 1) < (1 + conn_latency) * (conn_interval_max >> 1)) {
```

Comment:
Doesn't this introduce a rounding error on conn. interval?

Say: STO=10, LAT=5 and CI=7
100 > 6 * 7 * 2.5 = 105
vs
20 > 6 * 3 = 18

Reply:
Instead of re-inventing, maybe just re-use the host expression?
```
((supervision_timeout  * 4U) <= ((1U + conn_latency) * conn_interval_max))
```

Reply:
Sure, I can add a partial fix for that

Reply:
@thoh-ot I've added a compensation for the rounding error.

Reply:
>Instead of re-inventing, maybe just re-use the host expression?

I missed this comment, sure that makes sense. Thanks @joerchan

Reply:
Done now, thanks both!